src/main/scala/org/hammerlab/guacamole/readsets/ReadSets.scala

```diff
@@ -0,0 +1,73 @@
+package org.hammerlab.guacamole.readsets
+
+import org.apache.spark.SparkContext
+import org.bdgenomics.adam.models.SequenceDictionary
+import org.hammerlab.guacamole._
+import org.hammerlab.guacamole.reads.Read
+
+case class ReadSets(readsRDDs: PerSample[ReadsRDD],
+                    sequenceDictionary: SequenceDictionary,
+                    contigLengths: ContigLengths) extends PerSample[ReadsRDD] {
+  override def length: Int = readsRDDs.length
+  override def apply(idx: Int): ReadsRDD = readsRDDs(idx)
+
+  val mappedReads = readsRDDs.map(_.mappedReads)
```

Comment:
We definitely don't need to worry about this causing any Spark jobs to run that wouldn't otherwise, and the RDD objects created here are tinydata, whether on the driver or executors. 

When I'm reading code and see a `lazy` qualifier, it signals to me that it is actually important that that `val` is only computed when it is used, presumably because it's expensive to compute and/or is frequently not used on instances of the class; I don't think either of those are the case here, so I dropped it.

I'm meaning to put a unit test somewhere that explores in depth how `lazy` behaves with Java and Kryo serialization; if it's important to you I can do that test before we merge this, though even if `lazy` doesn't come with any serde gotchas, I think I prefer this way, for the reasons discussed above.